hazelcast/src/main/java/com/hazelcast/cluster/impl/TcpIpJoiner.java

```diff
@@ -55,12 +59,27 @@
 
 public class TcpIpJoiner extends AbstractJoiner {
 
+    /**
+     * Specifies how long the address of a member that has previously joined the cluster
+     * will be retained in the knownMemberAddresses after the member leaves the cluster.
+     */
+    public static final String PREVIOUSLY_JOINED_MEMBER_ADDRESS_RETENTION_DURATION_PROP =
+            "hazelcast.previously.joined.member.address.retention.duration.seconds";
+    // Selected this default value same as default value of missing-cp-member-auto-removal-seconds
+    private static final int DEFAULT_PREVIOUSLY_JOINED_MEMBER_ADDRESS_RETENTION_DURATION_IN_SECS = 14400;
     private static final long JOIN_RETRY_WAIT_TIME = 1000L;
     private static final int MASTERSHIP_CLAIM_TIMEOUT = 10;
 
     private final int maxPortTryCount;
     private volatile boolean claimingMastership;
     private final JoinConfig joinConfig;
+    private final long previouslyJoinedMemberAddressRetentionDuration;
```

Comment:
Maybe we can use Duration class instead? Map then could store <Address, Instant> and comparison would be just:
Duration.between(Instant.now(), iterator.next()).compareTo(previouslyJoinedMemberAddressRetentionDuration) > 0

Reply:
Since IMDG 3.12.z is Java 6 compatible, I couldn't make this change. Maybe we can discuss this change on the main 5.2 change.

Reply:
Right, true